syntax = "proto3";

package myservice;

message NestedMessage {
  string inner_message = 1;
}

message MyRequest {
  string message = 1;
  repeated int32 numbers = 2;
  map<string, int32> key_value_pairs = 3;
  NestedMessage nested_message = 4;
}

message MyResponse {
  string message = 1;
  NestedMessage nested_message = 2;
}

service MyService {
  rpc SendMessage(MyRequest) returns (MyResponse);
}
